src/configure-webpack/loaders/javascript.js

```diff
@@ -50,7 +50,8 @@ export default {
                 // see: https://github.com/babel/babel/issues/4702
                 ...(action === actions.DEVELOP || action === actions.TEST_UNIT ? [
                   [require.resolve('babel-plugin-transform-react-jsx-source'), {}],
-                  [require.resolve('babel-plugin-transform-react-jsx-self'), {}]
+                  [require.resolve('babel-plugin-transform-react-jsx-self'), {}],
+                  [require.resolve('babel-plugin-syntax-dynamic-import'), {}]
```

Comment:
This shouldn't be only available in development mode. You could use the dynamic import feature for production builds as well.